src/wayland_frontend/engine.rs

```diff
@@ -52,8 +53,26 @@ impl Engine {
     pub fn new(coordinator: Coordinator,
                settings: Settings,
                keyboard_config: KeyboardConfig) -> Self {
+        let mut partial_socket_path = PathBuf::from(std::env::var("XDG_RUNTIME_DIR").unwrap_or("/run/user/1000".to_owned()));
+        let mut socket = None;
+        for i in 0..10{
+            partial_socket_path.push(format!("wayland-{}", i));
+            let try_sock = wl::DisplaySocket::new(&partial_socket_path);
+            if let Ok(sock) = try_sock{
+                socket = Some(sock);
+                break;
+            }
+            partial_socket_path.pop();
+        }
+        if let None = socket{
+            log_warn3!("wayland engine ERROR: cannot open a socket in the 0-9 range");
+            std::process::exit(1);
+        }
+        else{
+            log_warn1!("socket created at {:?}", partial_socket_path)
+        }
         Engine {
-            display: wl::DisplaySocket::new_default().expect("Creating display socket"),
+            display: socket.expect("Fatal  ERROR"),
```

Comment:
"Fatal ERROR" is too generic. This message should be informative enough to know immediately what went wrong when you see it in logs.